src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ComplexTypeModelBinder.cs

```diff
@@ -57,6 +57,11 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
             // Create model first (if necessary) to avoid reporting errors about properties when activation fails.
             if (bindingContext.Model == null)
             {
+                if (!HasDefaultConstructor(bindingContext.ModelType.GetTypeInfo()))
+                {
+                    return;
```

Comment:
1. Check of `Model != null || {has a default constructor}` should be done either as part of `CanCreateModel()` or after the call to that method in `BindModelCore()`. No need to do this after going `async`.
2. Need an error in `ModelState` or (probably better) an `Exception` when `Model == null && {doesn't have a default constructor}`. Suspect the code previously in the provider would lead to an `InvalidOperationException` because no model binder would match. I'd prefer something similar here (even though the binder runs later) because the problem isn't recoverable without a code change.